### ISSUE TITLE
DOM Level 3 deficiencies

### ISSUE THREAD

**Issue by GoogleCodeExporter** [classilla/classilla#40]
```
This affects even the front page of Google itself, so we need to fix as
much of this as possible for 9.0.4. This tracker will cover relevant
bugzilla numbers.

nsIDOM3Node.textContent (VERY IMPORTANT!)
https://bugzilla.mozilla.org/show_bug.cgi?id=210451
https://bugzilla.mozilla.org/show_bug.cgi?id=210556

XPath
https://bugzilla.mozilla.org/show_bug.cgi?id=218756

Lower Priority things:

Constants to API
https://bugzilla.mozilla.org/show_bug.cgi?id=245476

adoptNode() ... meh
https://bugzilla.mozilla.org/show_bug.cgi?id=330677
```

Original issue reported on code.google.com by `classi...@floodgap.com` on 8 Aug 2009 at 2:59
- Blocked on: #1
